geomstats/learning/radial_kernel_functions.py

```diff
@@ -0,0 +1,432 @@
+"""Radial kernel functions.
+
+References
+----------
+https://en.wikipedia.org/wiki/Kernel_(statistics)
+https://en.wikipedia.org/wiki/Radial_basis_function
+
+Notes
+-----
+We chose not to apply the normalization coefficients used in some references.
+These normalization coefficients are used in order that the kernel functions
+integrate to 1 on the Euclidean space of dimension 1.
+"""
+
+import geomstats.backend as gs
+
+
+def _check_distance(distance):
+    """Check if the distance if a non-negative real number."""
+    if gs.any(distance < 0):
+        raise ValueError('The distance should be a non-negative real number.')
+    distance = gs.array(distance, dtype=float)
+    return distance
+
+
+def _check_bandwidth(bandwidth):
+    """Check if the bandwidth is a positive real number."""
+    if gs.any(bandwidth <= 0):
+        raise ValueError('The bandwidth should be a positive real number.')
+    bandwidth = gs.array(bandwidth, dtype=float)
+    return bandwidth
+
+
+def uniform_radial_kernel(distance, bandwidth=1.0):
+    """Uniform radial kernel.
+
+    Parameters
+    ----------
+    distance : array-like
+        Array of non-negative real values.
+    bandwidth : float, optional (default=1.0)
+        Positive scale parameter of the kernel.
+
+    Returns
+    -------
+    weight : array-like
+        Array of non-negative real values of the same shape than
+        parameter 'distance'.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Kernel_(statistics)
+    """
+    distance = _check_distance(distance)
+    bandwidth = _check_bandwidth(bandwidth)
+    scaled_distance = distance / bandwidth
+    weight = gs.where(
+        scaled_distance < 1,
+        gs.ones(distance.shape, dtype=float),
+        gs.zeros(distance.shape, dtype=float))
+    return weight
+
+
+def triangular_radial_kernel(distance, bandwidth=1.0):
+    """Triangular radial kernel.
+
+    Parameters
+    ----------
+    distance : array-like
+        Array of non-negative real values.
+    bandwidth : float, optional (default=1.0)
+        Positive scale parameter of the kernel.
+
+    Returns
+    -------
+    weight : array-like
+        Array of non-negative real values of the same shape than
+        parameter 'distance'.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Kernel_(statistics)
+    """
+    distance = _check_distance(distance)
+    bandwidth = _check_bandwidth(bandwidth)
+    scaled_distance = distance / bandwidth
+    weight = gs.where(
+        scaled_distance < 1,
+        1 - scaled_distance,
+        gs.zeros(distance.shape, dtype=float))
```

Comment:
You might not need this cast to dtype=float with the new version of `gs.where` of the PR merged yesterday (it should do the cast automatically). Could you check? Thanks